promql/engine.go

```diff
@@ -207,13 +207,29 @@ func contextErr(err error, env string) error {
 	}
 }
 
+// QueryTracker provides access to two features:
+// 1) tracking of active query that are logged on restart if PromQL engine crashes while executing the query, and
+// 2) enforcement of the maximum number of concurrent requests.
```

Comment:
(please break this into multiple lines where appropriate)
```suggestion
// QueryTracker provides access to two features:
// 1) Tracking of active query. If PromQL engine crashes while executing any query, they should be present in the tracker on restart, hence logged. After the logging on restart, the tracker gets emptied.
// 2) Enforcement of the maximum number of concurrent queries.
```